src/Avalonia.Controls/Window.cs

```diff
@@ -239,7 +244,43 @@ protected override void HandleApplicationExiting()
         public void Close(object dialogResult)
         {
             _dialogResult = dialogResult;
-            Close();
+            Close(false);
+        }
+
+        internal void Close(bool ignoreCancel)
+        {
+            bool cancelClosing;
+            try
+            {
+                cancelClosing = HandleClosing();
+            }
+            catch
+            {
+                CloseWindow();
```

Comment:
Might be better to transform it to a `finally` block, so debugger won't treat the exception as handled.

Reply:
Done.